pkg/fieldpath/fieldpath.go

```diff
@@ -18,26 +18,38 @@ package fieldpath
 
 import (
 	"fmt"
+	"sort"
+	"strconv"
 	"strings"
 
 	"k8s.io/apimachinery/pkg/api/meta"
-	"k8s.io/apimachinery/pkg/util/sets"
 	"k8s.io/apimachinery/pkg/util/validation"
 )
 
 // FormatMap formats map[string]string to a string.
 func FormatMap(m map[string]string) (fmtStr string) {
 	// output with keys in sorted order to provide stable output
-	keys := sets.NewString()
-	for key := range m {
-		keys.Insert(key)
+	keys := make([]string, 0, len(m))
+	var grow int
+	for k, v := range m {
+		keys = append(keys, k)
+		// why + 2? (=,\n)
+		// this is just a vague size to avoid frequent requests for space
+		// why is it 3*len(v)/2, from strconv.quoteWith
```

Comment:
I don't understand the `3*len(v)/2` even after this comment. Can you expand?

Reply:
I'm not convinced we need this estimation. This function is used to append labels, whose values don't need escaping, and annotations, whose values usually don't need escaping.

Allocating a buffer with enough space for len(k) + len(v) + 3 (for `=`, `"` and `"`) and writing into the buffer would be much simpler to implement and read.